### PR TITLE
Warning when NoSync setting is set as it does not turn syncing off…

### PR DESCRIPTION
**Pull request by eginez**
… #3799

### CODE REVIEW

File: repo/fsrepo/fsrepo.go [hunk 1]
```diff
@@ -400,6 +400,9 @@ func (r *FSRepo) openDatastore() error {
 	} else if r.config.Datastore.Spec == nil {
 		return fmt.Errorf("required Datastore.Spec entry missing from config file")
 	}
+	if r.config.Datastore.NoSync {
+		log.Warning("NoSync is no longer effective. Set 'sync' to false. See https://github.com/ipfs/go-ipfs/blob/master/docs/datastores.md#flatfs.")
```
Comment (whyrusleeping) [inline repo/fsrepo/fsrepo.go]: Hrm... I think the wording might be clearer as:
"NoSync is now deprecated in favor of datastore specific settings. If you want to disable fsync on flatfs 'sync' to false,  See <that link>"

**Comment by whyrusleeping** [comment]
Thanks for getting to the root of the problem :)

**Comment by eginez** [comment]
Hey @whyrusleeping I think this should be ready to be merged

**Comment by whyrusleeping** [comment]
Thanks for the ping! and the PR!

**Comment by lgierth** [comment]
Do I understood the linked issue #3799 right, we ignore Datastore.NoSync?

**Comment by eginez** [comment]
That's correct Datastore.NoSync is ignored

**Pull request merged**